### DISCUSSION TITLE
Additional UI Widget types

### DISCUSSION THREAD

**Discussion by schwehr** [Textualize/textual#1241] [Q&A] [answered]
More a question than an issue.  Sorry if this is the wrong place to ask.

Context: I'm looking to make a small textual app that is basically a guided database entry tool / form entry or to modify a pre-existing entry.  Specifically it's for [SpatioTemporal Asset Catalog (STAC)](https://github.com/radiantearth/stac-spec) and one concrete example: [Earth Engine STAC Catalog](https://radiantearth.github.io/stac-browser/#/external/earthengine-stac.storage.googleapis.com/catalog/catalog.json).  Not that it really matters, but the results will be stored in a JSON file.

Before I go and try to implement some of these UI widgets, I wanted to make sure I'm not missing something that is already done.

https://en.wikipedia.org/wiki/Graphical_widget

Some of the widgets I'm looking at doing:

- Input box constrained by a type and range, e.g.
  - Text that must match a regex
  - Int or float in a range
- A spinner to select from a list of choices, e.g. pick a license from https://spdx.org/licenses/)

Also, does anyone have examples of textual applications like this?

**Comment by davep** [comment, marked as answer]
There isn't anything built into Textual right at the moment that would directly satisfy those particular requirements, but I imagine extending one or more of the existing widgets like this should work. You'll also find some of this [on the roadmap](https://textual.textualize.io/roadmap/).
